### ISSUE TITLE
Wrong size of mask in RTMDet-Ins post-processing

### ISSUE THREAD

**Issue by flytocc** [open-mmlab/mmdetection#11230]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
In `_bbox_mask_post_process` function, when the argument `rescale` is True, the `mask_logits` will be interpolated to the original image size, but the width and height in `scale_factor` are reversed when calculating the original image size. If `keep_ratio` in `Resize` tranform is False, the width and height scale factor in `scale_factor` is difference, and it will make mask size wrong.

https://github.com/open-mmlab/mmdetection/blob/ee2e5423571d57d7f4156bade5a2cd6470fdffae/mmdet/models/dense_heads/rtmdet_ins_head.py#L498-L507

**Reproduction**

1. What command or script did you run?

```none
python tools/test.py configs/rtmdet/rtmdet-ins_tiny_8xb32-300e_coco.py \
    https://download.openmmlab.com/mmdetection/v3.0/rtmdet/rtmdet-ins_tiny_8xb32-300e_coco/rtmdet-ins_tiny_8xb32-300e_coco_20221130_151727-ec670f7e.pth \
    --cfg-options test_dataloader.dataset.pipeline.1.keep_ratio=False
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?

There are no modifications on the code or config.

3. What dataset did you use?

coco

**Environment**

not important

**Error traceback**
If applicable, paste the error trackback here.

```none
none
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

https://github.com/open-mmlab/mmdetection/pull/11231

**Comment by flytocc** [comment]
@RangiLyu Hi, can you take a look at this issue, the same error is fixed in mmdeploy https://github.com/open-mmlab/mmdeploy/pull/2581
